test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/EmailPasswordAuthTests.kt

```diff
@@ -53,511 +68,225 @@ class EmailPasswordAuthTests {
-//            try {
-//                when (method) {
-//                    Method.REGISTER_USER -> provider.registerUser(email, "123456")
-//                    Method.CONFIRM_USER -> provider.confirmUser("token", "tokenId")
-//                    Method.RESEND_CONFIRMATION_EMAIL -> provider.resendConfirmationEmail(email)
-//                    Method.SEND_RESET_PASSWORD_EMAIL -> provider.sendResetPasswordEmail(email)
-//                    Method.CALL_RESET_PASSWORD_FUNCTION -> provider.callResetPasswordFunction(email, "123456")
-//                    Method.RETRY_CUSTOM_CONFIRMATION -> provider.retryCustomConfirmation(email)
-//                    Method.RESET_PASSWORD -> provider.resetPassword("token", "token-id", "password")
-//                }
-//                Assert.fail("$method should have thrown an exception")
-//            } catch (error: AppException) {
-//                assertEquals(ErrorCode.NETWORK_UNKNOWN, error.errorCode)
-//            }
-//        }
-//    }
-//
-//    @Test
-//    fun callAsyncMethodsOnNonLooperThreadThrows() {
-//        val provider: EmailPasswordAuth = app.emailPassword
-//        val email: String = TestHelper.getRandomEmail()
-//        val callback = App.Callback<Void> { Assert.fail() }
-//        for (method in Method.values()) {
-//            try {
-//                when (method) {
-//                    Method.REGISTER_USER -> provider.registerUserAsync(email, "123456", callback)
-//                    Method.CONFIRM_USER -> provider.confirmUserAsync("token", "tokenId", callback)
-//                    Method.RESEND_CONFIRMATION_EMAIL -> provider.resendConfirmationEmailAsync(email, callback)
-//                    Method.SEND_RESET_PASSWORD_EMAIL -> provider.sendResetPasswordEmailAsync(email, callback)
-//                    Method.CALL_RESET_PASSWORD_FUNCTION -> provider.callResetPasswordFunctionAsync(email, "123456", arrayOf(), callback)
-//                    Method.RETRY_CUSTOM_CONFIRMATION -> provider.retryCustomConfirmationAsync(email, callback)
-//                    Method.RESET_PASSWORD -> provider.resetPasswordAsync("token", "token-id", "password", callback)
-//                }
-//                Assert.fail("$method should have thrown an exception")
-//            } catch (ignore: IllegalStateException) {
-//            }
-//        }
-//    }
+    @Test
+    fun confirmUser_invalidArgumentsThrows() {
+        val provider = app.emailPasswordAuth
+        runBlocking {
+            assertFailsWith<IllegalArgumentException> { provider.confirmUser("", "token-id") }
+            assertFailsWith<IllegalArgumentException> { provider.confirmUser("token", "") }
+        }
+    }
+
+    @Test
+    fun resendConfirmationEmail() = runBlocking {
+        // We only test that the server successfully accepts the request. We have no way of knowing
+        // if the Email was actually sent.
+        // TODO Figure out a way to check if this actually happened. Perhaps a custom SMTP server?
```

Comment:
One way could be to set up a custom validation function on Atlas/Stitch that will send a response to the node instance.

Reply:
Good idea, but probably not worth the effort right now. Will defer to a future PR.